lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -1087,564 +1120,76 @@ def _on_backup_relation_broken(self, event: RelationEvent) -> None:  # noqa: C90
         self.charm.status.clear(BackupInDisabling)
         self.charm.status.clear(PluginConfigError)
 
-    def _execute_s3_broken_calls(self):
-        """Executes the s3 broken API calls."""
-        return  # do not execute anything as we intend to keep the backups untouched
 
-    def _check_repo_status(self) -> BackupServiceState:
-        try:
-            response = self.charm.opensearch.request(
-                "GET",
-                f"_snapshot/{S3_REPOSITORY}",
-                retries=6,
-                timeout=10,
-            )
-            return self.get_service_status(response)
-        except OpenSearchHttpError:
-            return BackupServiceState.RESPONSE_FAILED_NETWORK
+class OpenSearchS3Backup(OpenSearchMainBackup):
```

Comment:
Amazing! I agree this is the very minimum they differ on, just the client lib and the plugin initialization. Very neat :)